src/py/flwr/client/app.py

```diff
@@ -400,6 +407,10 @@ def _on_backoff(retry_state: RetryState) -> None:
                         else:
                             runs[run_id] = Run(run_id, "", "", {})
 
+                    # NodeState must be initialized at this point
+                    if node_state is None:
+                        raise ValueError("NodeState is not initialized")
```

Comment:
It seems this is ever going to be hit? because there will be a `ValueError` earlier in the code if `node_id` is `None` or if the transport doesn't have such concept.

Reply:
True. We could remove it. Not sure if the linters would allow it, though.